File: DotNetMessaging/backend/DotNetMessaging.API/Models/MessageReaction.cs
using MongoDB.Bson.Serialization.Attributes;

namespace DotNetMessaging.API.Models;

public class MessageReaction : BaseEntity
{
    [BsonElement("messageId")]
    public string MessageId { get; set; } = string.Empty;

    [BsonElement("userId")]
    public string UserId { get; set; } = string.Empty;

    [BsonElement("emoji")]
    public string Emoji { get; set; } = string.Empty;
}

File: DotNetMessaging/SpringBootMessaging/src/main/java/com/messaging/api/models/MessageReaction.java
package com.messaging.api.models;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@EqualsAndHashCode(callSuper = true)
@Document(collection = "messageReactions")
public class MessageReaction extends BaseEntity {
    private String messageId;
    
    private String userId;
    
    private String emoji;
}

